src/components/chatBubble/chatBubble.tsx

```diff
@@ -1,16 +1,34 @@
 import './styles.css'
+import CodeBubble from "../codeBubble/codeBubble"
+import Markdown from 'react-markdown';
 
 interface Props {
     text: string,
     chat_style: string
 }
 
 const ChatBubble: React.FC<Props> = ({ text, chat_style }) => {
+    let codeChunks = text.match(/```[^```]+```/g) || []
+
+    let textChunks = text.split(/```[^```]+```/)
```

Comment:
What is this regex looking for?

Reply:
The text regex matches all chunks of the response that aren't code chunks. Code chunks being text wrapped in 3 backticks. The approach with these regexes is somewhat a shortcut because if Magnus's response includes code whatsoever the pattern will always be ever other "chunk" is code. the first and last chunks will always be text. These regexes aren't perfect and need some tweaking tho